# Week 3 - Lab 2 

## Loading Data with Pandas 

## A.) View the Data 

In [5]:
import warnings
warnings.simplefilter('ignore')

import pandas as pd

df = pd.read_csv('s3://ia340-2025-fall-67/Meteorite_Landings.csv') # load the data from s3
df.head() #show the top 5 rows; using tail shows the bottom 5 

name   id nametype     recclass  mass (g)  fall    year    reclat  \
0    Aachen    1    Valid           L5      21.0  Fell  1880.0  50.77500   
1    Aarhus    2    Valid           H6     720.0  Fell  1951.0  56.18333   
2      Abee    6    Valid          EH4  107000.0  Fell  1952.0  54.21667   
3  Acapulco   10    Valid  Acapulcoite    1914.0  Fell  1976.0  16.88333   
4   Achiras  370    Valid           L6     780.0  Fell  1902.0 -33.16667   

     reclong           GeoLocation  
0    6.08333     (50.775, 6.08333)  
1   10.23333  (56.18333, 10.23333)  
2 -113.00000    (54.21667, -113.0)  
3  -99.90000     (16.88333, -99.9)  
4  -64.95000   (-33.16667, -64.95)

In [22]:
df.shape #show the number of rows and columns and check the dataset size

(38115, 10)

In [14]:
df.info() # shows the data types and helps detect missing daa and confirm correct formats

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45716 entries, 0 to 45715
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   name         45716 non-null  object 
 1   id           45716 non-null  int64  
 2   nametype     45716 non-null  object 
 3   recclass     45716 non-null  object 
 4   mass (g)     45585 non-null  float64
 5   fall         45716 non-null  object 
 6   year         45425 non-null  float64
 7   reclat       38401 non-null  float64
 8   reclong      38401 non-null  float64
 9   GeoLocation  38401 non-null  object 
dtypes: float64(4), int64(1), object(5)
memory usage: 3.5+ MB


In [15]:
df.describe() #gives you a statitical summary 

id      mass (g)          year        reclat       reclong
count  45716.000000  4.558500e+04  45425.000000  38401.000000  38401.000000
mean   26889.735104  1.327808e+04   1991.828817    -39.122580     61.074319
std    16860.683030  5.749889e+05     25.052766     46.378511     80.647298
min        1.000000  0.000000e+00    860.000000    -87.366670   -165.433330
25%    12688.750000  7.200000e+00   1987.000000    -76.714240      0.000000
50%    24261.500000  3.260000e+01   1998.000000    -71.500000     35.666670
75%    40656.750000  2.026000e+02   2003.000000      0.000000    157.166670
max    57458.000000  6.000000e+07   2101.000000     81.166670    354.473330

### In my dataset there is 38115 rows and 10 columns. The columns include: name (object), id (interger), nametype (object), recclass (object), mass (g) (float), fall (object), year (float), reclat (float), reclong (float), geolocation (object). My first impression from viewing my data was how many meterorites have been recorded from NASA. 

## B. Clean the Data 

In [ ]:
df.rename

In [9]:
df.sort_values(by='mass (g)') #order rows by column values 

name     id nametype      recclass  mass (g)   fall  \
31063       Österplana 050  56149   Relict     Relict OC       0.0  Found   
25557  Miller Range 090478  55953    Valid           CO3       0.0  Found   
31076       Österplana 063  56162   Relict     Relict OC       0.0  Found   
12640                 Gove  52859   Relict   Relict iron       0.0  Found   
31067       Österplana 054  56153   Relict     Relict OC       0.0  Found   
...                    ...    ...      ...           ...       ...    ...   
38282       Wei-hui-fu (a)  24231    Valid          Iron       NaN  Found   
38283       Wei-hui-fu (b)  24232    Valid          Iron       NaN  Found   
38285              Weiyuan  24233    Valid  Mesosiderite       NaN  Found   
41472        Yamato 792768  28117    Valid           CM2       NaN  Found   
45698        Zapata County  30393    Valid          Iron       NaN  Found   

         year    reclat    reclong             GeoLocation  
31063  2003.0  58.58333   13.43333    (58.58333, 13.43333)  
25557  2009.0   0.00000    0.00000              (0.0, 0.0)  
31076  2010.0  58.58333   13.43333    (58.58333, 13.43333)  
12640  1979.0 -12.26333  136.83833  (-12.26333, 136.83833)  
31067  2005.0  58.58333   13.43333    (58.58333, 13.43333)  
...       ...       ...        ...                     ...  
38282  1931.0       NaN        NaN                     NaN  
38283  1931.0       NaN        NaN                     NaN  
38285  1978.0  35.26667  104.31667   (35.26667, 104.31667)  
41472  1979.0 -71.50000   35.66667       (-71.5, 35.66667)  
45698  1930.0  27.00000  -99.00000           (27.0, -99.0)  

[45716 rows x 10 columns]

In [8]:
df['mass (g)'] #select a column

0            21.0
1           720.0
2        107000.0
3          1914.0
4           780.0
           ...   
45711       172.0
45712        46.0
45713         3.3
45714      2167.0
45715       200.0
Name: mass (g), Length: 45716, dtype: float64

In [13]:
df[df['mass (g)'] > 10000000] #shows onlu rows where mass is > 10000000

name     id nametype         recclass    mass (g)   fall  \
920       Sikhote-Alin  23593    Valid       Iron, IIAB  23000000.0   Fell   
3455           Armanty   2335    Valid       Iron, IIIE  28000000.0  Found   
5016        Bacubirito   4919    Valid  Iron, ungrouped  22000000.0  Found   
5365   Campo del Cielo   5247    Valid     Iron, IAB-MG  50000000.0  Found   
5370     Canyon Diablo   5257    Valid     Iron, IAB-MG  30000000.0  Found   
5373         Cape York   5262    Valid      Iron, IIIAB  58200000.0  Found   
5468        Chupaderos   5363    Valid      Iron, IIIAB  24300000.0  Found   
12613           Gibeon  10912    Valid        Iron, IVA  26000000.0  Found   
16392             Hoba  11890    Valid        Iron, IVB  60000000.0  Found   
22921            Mbosi  15456    Valid  Iron, ungrouped  16000000.0  Found   
26174           Morito  16745    Valid      Iron, IIIAB  10100000.0  Found   
26297      Mundrabilla  16852    Valid    Iron, IAB-ung  24000000.0  Found   
38317       Willamette  24269    Valid      Iron, IIIAB  15500000.0  Found   

         year    reclat    reclong             GeoLocation  
920    1947.0  46.16000  134.65333      (46.16, 134.65333)  
3455   1898.0  47.00000   88.00000            (47.0, 88.0)  
5016   1863.0  26.20000 -107.83333      (26.2, -107.83333)  
5365   1575.0 -27.46667  -60.58333  (-27.46667, -60.58333)  
5370   1891.0  35.05000 -111.03333     (35.05, -111.03333)  
5373   1818.0  76.13333  -64.93333   (76.13333, -64.93333)  
5468   1852.0  27.00000 -105.10000          (27.0, -105.1)  
12613  1836.0 -25.50000   18.00000           (-25.5, 18.0)  
16392  1920.0 -19.58333   17.91667   (-19.58333, 17.91667)  
22921  1930.0  -9.11667   33.06667    (-9.11667, 33.06667)  
26174  1600.0  27.05000 -105.43333     (27.05, -105.43333)  
26297  1911.0 -30.78333  127.55000     (-30.78333, 127.55)  
38317  1902.0  45.36667 -122.58333  (45.36667, -122.58333)

## 3.) Explore and Clean Data 

In [ ]:
df.datetime

In [17]:
df.isnull().sum #shows the number of rows containing null value

<bound method DataFrame.sum of         name     id  nametype  recclass  mass (g)   fall   year  reclat  \
0      False  False     False     False     False  False  False   False   
1      False  False     False     False     False  False  False   False   
2      False  False     False     False     False  False  False   False   
3      False  False     False     False     False  False  False   False   
4      False  False     False     False     False  False  False   False   
...      ...    ...       ...       ...       ...    ...    ...     ...   
45711  False  False     False     False     False  False  False   False   
45712  False  False     False     False     False  False  False   False   
45713  False  False     False     False     False  False  False   False   
45714  False  False     False     False     False  False  False   False   
45715  False  False     False     False     False  False  False   False   

       reclong  GeoLocation  
0        False        False  
1       

In [18]:
df = df.dropna() #drop rows that contain missing data and prdouces a clean dataset 
df.isnull().sum()

name           0
id             0
nametype       0
recclass       0
mass (g)       0
fall           0
year           0
reclat         0
reclong        0
GeoLocation    0
dtype: int64

In [21]:
df['mass (g)'].value_counts() #shows the frequency of each unique alue in column 

mass (g)
1.30      155
1.20      122
1.40      120
2.10      115
2.40      108
         ... 
27.69       1
15.98       1
632.70      1
18.87       1
21.87       1
Name: count, Length: 11248, dtype: int64

### In my dataset there are 45716 rows and 10 columns.